### PR TITLE
feat(examples): flexible bootstrap options

### LINKED ISSUE #11: Add additional bootstrap parameters to the vpc_peering_dedicated example
### Is your feature request related to a problem?

Current GCP examples don't support `dgname`, `tplname` and `cgname` at example level and requires modifications to the main.tf file. 

https://github.com/PaloAltoNetworks/terraform-google-swfw-modules/blob/2030253c2eb7cca1e383f501af375b245b094e22/examples/vpc_peering_dedicated/main.tf#L37-L42

### Describe the solution you'd like

Expand the bootstrap template files to include the extra bootstrap parameters (`dgname`, `tplname` and `cgname`)

### Describe alternatives you've considered.

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by michalbil**
## Description

Modify the way bootstrap options are passed in examples, so that any required one can be added at common/instance level, as needed.

## Motivation and Context

Resolves #11 

## How Has This Been Tested?

Run plan/apply one of examples ("multi_nic_common") example with extended bootstrap options, on both common and instance levels.

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes if appropriate.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: examples/multi_nic_common/main.tf [hunk 1]
```diff
@@ -32,17 +32,14 @@ resource "local_file" "bootstrap_xml" {
...
-  content = templatefile("templates/init-cfg.tmpl",
-    {
-      panorama-server = try(each.value.bootstrap_options.panorama-server, var.vmseries_common.bootstrap_options.panorama-server, "")
-      type            = try(each.value.bootstrap_options.type, var.vmseries_common.bootstrap_options.type, "")
-      dns-primary     = try(each.value.bootstrap_options.dns-primary, var.vmseries_common.bootstrap_options.dns-primary, "")
-      dns-secondary   = try(each.value.bootstrap_options.dns-secondary, var.vmseries_common.bootstrap_options.dns-secondary, "")
-  })
+  content = templatefile(
+    "templates/init-cfg.tmpl",
+    { bootstrap_options = merge(var.vmseries_common.bootstrap_options, each.value.bootstrap_options) }
```
Comment (pavelrn) [inline examples/multi_nic_common/main.tf:41]: Should we exclude some bootstrap_options here like `authcodes` for example?

File: examples/multi_nic_common/main.tf [hunk 2]
```diff
@@ -56,8 +53,8 @@ module "bootstrap" {
   service_account = module.iam_service_account[each.value.service_account_key].email
   location        = each.value.location
   files = merge(
-    { for k, v in var.vmseries : "files/${k}/config/bootstrap.xml" => "${k}/config/bootstrap.xml" },
-    { for k, v in var.vmseries : "files/${k}/config/init-cfg.txt" => "${k}/config/init-cfg.txt" },
+    { for k, v in var.vmseries : "files/${k}/config/bootstrap.xml" => "${k}/config/bootstrap.xml" if can(v.bootstrap_template_map) },
```
Comment (pavelrn) [inline examples/multi_nic_common/main.tf:56]: How do we handle `authcodes` in the example? How we do the licensing in general?

File: examples/vmseries_ha/main.tf [hunk 3]
```diff
@@ -35,17 +35,14 @@ resource "local_file" "bootstrap_xml" {
 resource "local_file" "init_cfg" {
 
   for_each = { for k, v in var.vmseries : k => v
-    if can(v.bootstrap_template_map)
+    if can(v.bootstrap_options)
```
Comment (pavelrn) [inline examples/vmseries_ha/main.tf:38]: I'm confused about which approach are we using to initialize VM-Series in the example: using the init-cfg.txt and the bucket or using VM-Series VM metadata?

Comment (pavelrn) [reply examples/vmseries_ha/main.tf:38]: Should we introduce some flag like `use_bootstrap_bucket` to select the bootstrap method?

File: examples/multi_nic_common/main.tf (hunk 2 above)
Comment (michalbil) [reply examples/multi_nic_common/main.tf:56]: Currently authcodes are not handled in the examples.

**Review by migara — APPROVED** [review]

**Pull request merged**